langchain/agents/tools/requests.ts

```diff
@@ -0,0 +1,69 @@
+import { Tool, Toolkit } from "./base";
+
+type headers = { [key: string]: string };
+
+interface RequestTool {
+  headers: headers;
+}
+
+export class RequestsGetTool extends Tool implements RequestTool {
+  name = "requests_get";
+
+  headers: headers;
+
+  constructor(headers?: headers) {
+    super();
+    this.headers = headers ?? {};
+  }
+
+  async call(input: string) {
+    const res = await fetch(input, {
+      headers: this.headers,
+    });
+    return res.text();
+  }
+
+  description = `A portal to the internet. Use this when you need to get specific content from a website. 
+  Input should be a  url (i.e. https://www.google.com). The output will be the text response of the GET request.`;
+}
+
+export class RequestsPostTool extends Tool implements RequestTool {
+  name = "requests_post";
+
+  headers: headers;
+
+  constructor(headers?: headers) {
+    super();
+    this.headers = headers ?? {};
+  }
+
+  async call(input: string) {
+    try {
+      const { url, data } = JSON.parse(input);
+      const res = await fetch(url, {
+        method: "POST",
+        headers: this.headers,
+        body: JSON.stringify(data),
+      });
+      return res.text();
+    } catch (error) {
+      return `${error}`;
+    }
+  }
+
+  description = `Use this when you want to POST to a website.
+  Input should be a json string with two keys: "url" and "data".
+  The value of "url" should be a string, and the value of "data" should be a dictionary of 
+  key-value pairs you want to POST to the url.
+  Be careful to always use double quotes for strings in the json string
+  The output will be the text response of the POST request.`;
+}
+
+export class RequestsToolkit extends Toolkit {
+  tools: Tool[];
+
+  constructor(headers?: headers) {
+    super();
+    this.tools = [new RequestsGetTool(headers), new RequestsPostTool(headers)];
```

Comment:
Why not make a more complete RequestsTool that includes the ability to consult the api spec (openapi)?

Reply:
Yeah can try and consolidate if people find useful. The goal is to create an interesting example for now